panacea/grpc_client.go

```diff
@@ -39,17 +50,39 @@ func makeInterfaceRegistry() sdk.InterfaceRegistry {
 	interfaceRegistry := sdk.NewInterfaceRegistry()
 	std.RegisterInterfaces(interfaceRegistry)
 	authtypes.RegisterInterfaces(interfaceRegistry)
-	markettypes.RegisterInterfaces(interfaceRegistry)
+	datadealtypes.RegisterInterfaces(interfaceRegistry)
+	datapooltypes.RegisterInterfaces(interfaceRegistry)
 	return interfaceRegistry
 }
 
-func (c *GrpcClient) Close() {
+func (c *GrpcClient) Close() error {
 	log.Info("closing Panacea gRPC connection")
-	c.conn.Close()
+	err := c.conn.Close()
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // GetPubKey gets the public key from blockchain.
-func (c *GrpcClient) GetPubKey(panaceaAddr string) ([]byte, error) {
+func (c *GrpcClient) GetPubKey(panaceaAddr string) (types.PubKey, error) {
+	client := authtypes.NewQueryClient(c.conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	response, err := client.Account(ctx, &authtypes.QueryAccountRequest{Address: panaceaAddr})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account info via grpc: %w", err)
+	}
+
+	var acc authtypes.AccountI
+	if err := c.interfaceRegistry.UnpackAny(response.GetAccount(), &acc); err != nil {
+		return nil, fmt.Errorf("failed to unpack account info: %w", err)
+	}
+	return acc.GetPubKey(), nil
```

Comment:
I think there is duplicate code. How about changing it like this?
```suggestion
	acc, err := c.GetAccount(panaceaAddr)
	if err != nil {
		return nil, err 
	}
	return acc.GetPubKey(), err
```

Reply:
I agree with your review. It could minimize communication with Panacea. Thank you.